packages/client/screens/WalletOverviewScreen.tsx

```diff
@@ -0,0 +1,120 @@
+import { Pressable, Box, Text, HStack, VStack, Center } from "native-base";
+import { StyleSheet } from "react-native";
+import { faWalletCustom } from "../components/icons/faWalletCustom";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+import { View } from "../components/Themed";
+import { faEthereumCustom } from "../components/icons/faEthereumCustom";
+import { CompositeScreenProps } from "@react-navigation/native";
+
+type Props = CompositeScreenProps<
+    HomeStackScreenProps<"WalletOverviewScreen">,
+    SettingsStackScreenProps<"WalletOverviewScreen">
+>;
+
+export default function WalletOverviewScreen({ route, navigation }: Props) {
+    const { address, name, currencyType, balance } = route.params;
+
+    function formatWalletAddress(address: string): string {
+        return `${address.substring(0, 6)}...${address.substring(address.length - 4, address.length)}`;
+    }
+
+    return (
+        <View style={styles.view}>
+            <Box style={styles.walletDetailsWrapper}>
+                <VStack style={styles.walletDetails}>
+                    <HStack justifyContent="space-between">
+                        <VStack>
+                            <Text style={styles.walletName}>{name}</Text>
+                            <Box marginTop="2px"></Box>
+                            <Text style={styles.walletAddress}>{formatWalletAddress(address)}</Text>
+                        </VStack>
+                        <VStack>
+                            <FontAwesomeIcon icon={faEthereumCustom} style={styles.ethereumIcon} size={40} />
+                        </VStack>
+                    </HStack>
+                    <HStack alignItems="center">
+                        <VStack>
+                            <Box marginTop="40px" marginBottom="0"></Box>
+                            <Text style={styles.currencyType}>{currencyType}</Text>
+                            <Text style={styles.walletBalance}>{balance}</Text>
+                        </VStack>
+                    </HStack>
+                </VStack>
+            </Box>
+            <Box marginTop="20px" marginBottom="0"></Box>
+            <Center>
+                <Pressable
+                    style={styles.button}
+                    onPress={() => navigation.navigate("WalletDetailsScreen", { address, name, currencyType, balance })}
+                >
+                    <Box style={styles.walletIconBackground}>
+                        <FontAwesomeIcon icon={faWalletCustom} style={styles.walletIcon} size={20} />
+                    </Box>
+                </Pressable>
+                <Box marginTop="4px"></Box>
+                <Text style={styles.detailsText}>Details</Text>
+            </Center>
+        </View>
+    );
+}
+
+const styles = StyleSheet.create({
+    view: {
+        flex: 1,
+    },
+    ethereumIcon: {
+        color: "#3C3C3D",
+    },
+    walletDetailsWrapper: {
+        paddingLeft: 20,
+        paddingRight: 20,
```

Comment:
Can be simplified to

```suggestion
        paddingVertical: 20
```